File: src/main/java/codility/Distinct.java
package codility;

import java.util.HashSet;

import static utils.Utils.require;

public class Distinct {

    public static void main(String[] args) {

        require(3 == solve(new int[]{2, 1, 1, 2, 3, 1}));
    }

    private static int solve(int[] values) {
        HashSet<Integer> set = new HashSet<>();

        for (int value : values) set.add(value);

        return set.size();
    }
}

File: src/main/kotlin/codility/Distinct.kt
package codility

fun main(args: Array<String>) {

    require(3 == solve(arrayOf(2, 1, 1, 2, 3, 1)))
    require(4 == solve(arrayOf(3, 4, 1, 2, 3)))
}

private fun solve(values: Array<Int>) = values.toSet().size